client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/steps/use-steps.tsx

```diff
@@ -0,0 +1,147 @@
+import { useTranslate } from 'i18n-calypso';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { MaybeLink } from './maybe-link';
+import type { Task, Expandable } from '@automattic/launchpad';
+
+const removeDuplicatedSlashes = ( url: string ) => url.replace( /(https?:\/\/)|(\/)+/g, '$1$2' );
+
+const getPluginInstallationPage = ( fromUrl: string ) => {
+	if ( fromUrl !== '' ) {
+		return removeDuplicatedSlashes(
+			`${ fromUrl }/wp-admin/plugin-install.php?s=%2522migrate%2520guru%2522&tab=search&type=term`
+		);
+	}
+
+	return 'https://wordpress.org/plugins/migrate-guru/';
+};
+
+const getMigrateGuruPageURL = ( siteURL: string ) =>
+	removeDuplicatedSlashes( `${ siteURL }/wp-admin/admin.php?page=migrateguru` );
+
+const recordInstructionsLinkClick = ( linkname: string ) => {
+	recordTracksEvent( 'calypso_site_migration_instructions_link_click', {
+		linkname,
+	} );
+};
+
+interface Options {
+	fromUrl: string;
+}
+
+interface Step {
+	task: Task;
+	expandable?: Expandable;
+}
+
+type Steps = Step[];
+
+export const useSteps = ( { fromUrl }: Options ): Steps => {
+	const translate = useTranslate();
+
+	const steps = [
+		{
+			title: translate( 'Install the Migrate Guru plugin' ),
+			content: (
+				<p>
+					{ translate(
+						"First you'll need to install and activate the {{a}}Migrate Guru plugin{{/a}} on the site you want to migrate. Click {{strong}}Next{{/strong}} when you're ready.",
+						{
+							components: {
+								strong: <strong />,
+								a: (
+									<a
+										href={ getPluginInstallationPage( fromUrl ) }
+										target="_blank"
```

Comment:
For links that open in another tab, we should use the [`ExternalLink` component](https://wpcalypso.wordpress.com/devdocs/design/external-link).

Reply:
TIL! Thank you!
Updated in https://github.com/Automattic/wp-calypso/pull/92070/commits/25389e93c3f0a764fcd291a6967bbd3cd5ed5928

Reply:
Hmm...I'm not seeing the little arrow beside the link as I was expecting to.

Reply:
I didn't add that because we don't have them in the design, and in the middle of the text it seems a little weird.
I applied in the 3 links we have and took a screenshot to see what you think.

<img width="1271" alt="Screenshot 2024-07-11 at 15 55 54" src="https://github.com/Automattic/wp-calypso/assets/876340/f9822441-0aea-46ca-a3b3-a40c1138d62a">

Reply:
Hmm..yeah, it does look a bit weird. @fditrapani I thought it was standard practice to use this for external links, but perhaps not mid-sentence?

Reply:
I'm going ahead and merging this PR without any change here. But if @fditrapani confirms that we should add the external icons, I can raise another PR. 😉

Reply:
> make the case for numbers

Sorry, @fditrapani, I didn't get what you meant here.
Do you mean applying it everywhere using the `ExternalLink` component?

If it is, the component supports both ways (with or without the icon), where the default option is without the icon. So I think we just need to decide how to use the component in this screen.

Reply:
aah sorry, my bad. I misread the conversation and thought we were talking about the numbers :D 

The external links do look a little weird but they offer more clarity and are a standard practice. Not a big deal if we didn't include them in this round. I don't think it would hurt to add them at a later stage.

Reply:
So just confirming, it would be expected to have the icons in all of them like the screenshot I sent, right? If yes, I'll change it in my other coming PR. 😉

Reply:
Yeah, including the icons would be great thanks!

Reply:
Thank you! I'll include them in the next PR.

Reply:
Added here with a screenshot: https://github.com/Automattic/wp-calypso/pull/92387/files#r1676137764

I also improved a little the size of the icons, which made things a little better than my previous screenshot. ;)

Reply:
Very nice! thanks for taking care of that.